File: leetcode/python/1.two-sum.py
def twoSum(a, b):
    x = {}
    i = 0
    while i < len(a):
        diff = b - a[i]
        if diff in x:
            return [x[diff],i]
        x[a[i]] = i
        i += 1


a = [3, 4, 5]

if __name__ == '__main__':
    print(twoSum(a, 8))

File: leetcode/kotlin/1.TwoSum.kt
fun twoSum(a: IntArray, b:Int): List<Int?>? {
    var x = mutableMapOf<Int,Int>();
    for((i,v) in a.withIndex()){
        if(x.containsKey(b -a[i])){
            return listOf(x[b -a[i]],i)
        }
        x[v] = i
    }
    return null
}

fun main(){
    val arr = intArrayOf(3,4,1);
    val ans =twoSum(arr, 5)
    print(ans)
}